bokeh/io.py

```diff
@@ -1,946 +0,0 @@
-
-        title = "Bokeh Plot"
-
-    return filename, resources, title
-
-def _save_helper(obj, filename, resources, title):
-    html = file_html(obj, resources, title=title)
-
-    with io.open(filename, mode="w", encoding="utf-8") as f:
-        f.write(decode_utf8(html))
-
-def push_notebook(document=None, state=None, handle=None):
-    ''' Update Bokeh plots in a Jupyter notebook output cells with new data
-    or property values.
-
-    When working the the notebook, the ``show`` function can be passed the
-    argument ``notebook_handle=True``, which will cause it to return a
-    handle object that can be used to update the Bokeh output later. When
-    ``push_notebook`` is called, any property updates (e.g. plot titles or
-    data source values, etc.) since the last call to ``push_notebook`` or
-    the original ``show`` call are applied to the Bokeh output in the
-    previously rendered Jupyter output cell.
-
-    Several example notebooks can be found in the GitHub repository in
-    the :bokeh-tree:`examples/howto/notebook_comms` directory.
-
-    Args:
-
-        document (Document, optional) :
-            A :class:`~bokeh.document.Document` to push from. If None,
-            uses ``curdoc()``. (default: None)
-
-        state (State, optional) :
-            A :class:`State` object. If None, then the current default
-            state (set by ``output_file``, etc.) is used. (default: None)
-
-    Returns:
-        None
-
-    Examples:
-
-        Typical usage is typically similar to this:
-
-        .. code-block:: python
-
-            from bokeh.plotting import figure
-            from bokeh.io import output_notebook, push_notebook, show
-
-            output_notebook()
-
-            plot = figure()
-            plot.circle([1,2,3], [4,6,5])
-
-            handle = show(plot, notebook_handle=True)
-
-            # Update the plot title in the earlier cell
-            plot.title.text = "New Title"
-            push_notebook(handle=handle)
-
-    '''
-    if state is None:
-        state = _state
-
-    if not document:
-        document = state.document
-
-    if not document:
-        warnings.warn("No document to push")
-        return
-
-    if handle is None:
-        handle = state.last_comms_handle
-
-    if not handle:
-        warnings.warn("Cannot find a last shown plot to update. Call output_notebook() and show(..., notebook_handle=True) before push_notebook()")
-        return
-
-    to_json = document.to_json()
-    if handle.doc is not document:
-        msg = dict(doc=to_json)
-    else:
-        msg = compute_patch_between_json(handle.json, to_json)
-
-    handle.comms.send(json.dumps(msg))
-    handle.update(document, to_json)
-
-def reset_output(state=None):
-    ''' Clear the default state of all output modes.
-
-    Returns:
-        None
-
-    '''
-    _state.reset()
-
-def _remove_roots(subplots):
```

Comment:
i'm pretty sure this is dead code. has it been removed (I didn't see it grepping around)?

Reply:
Yup, removed it